src/parser/sparqlParser/SparqlQleverVisitor.h

```diff
@@ -1070,16 +1113,46 @@ class SparqlQleverVisitor : public SparqlAutomaticVisitor {
 
   antlrcpp::Any visitIriOrFunction(
       SparqlAutomaticParser::IriOrFunctionContext* ctx) override {
-    if (ctx->argList()) {
-      throw SparqlParseException{
-          "calls to non-built-in functions in expressions are not supported by "
-          "this parser"};
+    // Case 1: Just an IRI.
+    if (ctx->argList() == nullptr) {
+      return ExpressionPtr{
+          std::make_unique<sparqlExpression::StringOrIriExpression>(
+              ctx->getText())};
     }
-
-    return ExpressionPtr{
-        std::make_unique<sparqlExpression::StringOrIriExpression>(
-            ctx->getText())};
-    return visitChildren(ctx);
+    // Case 2: Function call, where the function name is an IRI.
+    auto iri = std::move(visitIri(ctx->iri()).as<std::string>());
```

Comment:
As you already stated above, this code is a duplicate.
There should be a function taking an `IRI` (as a string) and the arguments (a vector of `ExpressionPtr`) and returns the function or an error (exactly the logic of 'Case 2', but then you can also use it inside the `visitFunctionCall`.

Reply:
Getting the return type and argument right wasn't trivial, but eventually I managed :-)